src/descriptor.rs

```diff
@@ -175,24 +175,64 @@ impl DescriptorWriter<'_> {
     pub fn interface(&mut self, number: InterfaceNumber,
         interface_class: u8, interface_sub_class: u8, interface_protocol: u8) -> Result<()>
     {
-        match self.num_interfaces_mark {
-            Some(mark) => self.buf[mark] += 1,
-            None => return Err(UsbError::InvalidState),
+        self.interface2(
+            number,
+            device::DEFAULT_ALTERNATE_SETTING,
+            interface_class,
+            interface_sub_class,
+            interface_protocol,
+            None,
+        )
+    }
+
+    /// Writes a interface descriptor with a specific alternate setting and
+    /// interface string identifier.
+    ///
+    /// # Arguments
+    ///
+    /// * `number` - Interface number previously allocated with
+    ///   [`UsbBusAllocator::interface`](crate::bus::UsbBusAllocator::interface).
+    /// * `alternate_setting` - number of the alternate setting
+    /// * `interface_class` - Class code assigned by USB.org. Use `0xff` for vendor-specific devices
+    ///   that do not conform to any class.
+    /// * `interface_sub_class` - Sub-class code. Depends on class.
+    /// * `interface_protocol` - Protocol code. Depends on class and sub-class.
+    /// * `interface_string` - Index of string descriptor describing this interface
+
+    pub fn interface2(
```

Comment:
I'd prefer a more descriptive name, maybe `interface_alternate()`?

Reply:
Right, a more descriptive name than `interface2` would be fitting.

Reply:
What about the original one: ```interface_with_alternate_setting```?

Reply:
Sounds good but it may be too long.

Reply:
Sounds good to me. Ian?

Reply:
I change the code. What do you mean with 'lan'?

Reply:
What i meant is if Ian has any suggestions. Anyway the change sounds good to me :)

Reply:
`interface_ex`? Too Win32?

Reply:
Perhaps even `interface_alt` :thinking: